lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -0,0 +1,548 @@
+""" API v0 views. """
+
+import datetime
+import logging
+import pytz
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.urlresolvers import reverse
+from django.core.validators import validate_email
+from django.db import transaction
+from django.http import Http404
+from django.utils.http import urlquote_plus
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from ccx_keys.locator import CCXLocator
+from courseware.courses import get_course_by_id
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from lms.djangoapps.ccx.models import CcxFieldOverride, CustomCourseForEdX
+from lms.djangoapps.ccx.overrides import (
+    get_override_for_ccx,
+    override_field_for_ccx,
+)
+from .paginators import CCXAPIPagination
+from .serializers import CCXCourseSerializer
+
+log = logging.getLogger(__name__)
+TODAY = datetime.datetime.today  # for patching in tests
+
+def get_valid_course(course_id, is_ccx=False):
+    """
+    Helper function used to validate and get a course from a course_id string.
+    It works with both master and ccx course id.
+
+    Args:
+        course_id (str): A string representation of a Master or CCX Course ID.
+        is_ccx (bool): Flag to perform the right validation
+
+    Returns:
+        tuple: a tuple of course_object, course_key, error_code, http_status_code
+    """
+    if course_id is None:
+        if not is_ccx:
+            log.info('Master course ID not provided')
+            error_code = 'master_course_id_not_provided',
+        else:
+            log.info('CCX Course ID not provided')
+            error_code = 'ccx_course_id_not_provided'
+
+        return None, None, error_code, status.HTTP_400_BAD_REQUEST
+
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        log.info('Course ID string "%s" is not valid', course_id)
+        return None, None, 'course_id_not_valid', status.HTTP_400_BAD_REQUEST
+
+    if not is_ccx:
+        try:
+            course_object = get_course_by_id(course_key)
+            return course_object, course_key, None, None
+        except Http404:
+            log.info('Master Course with ID "%s" not found', course_id)
+            return None, None, 'course_id_does_not_exist', status.HTTP_404_NOT_FOUND
```

Comment:
The reason is because I need to send back a custom message that includes only the error code and I wanted to use a common pattern for all the error cases.